apps/cow-tools/src/app/milkman/[network]/order/new/page.tsx

```diff
@@ -252,86 +211,160 @@ function FormHeader({
   );
 }
 
-function FormSelectTokens({
-  form,
+function FormOrderOverview({
+  onSubmit,
   userAddress,
+  defaultValues,
 }: {
-  form: UseFormReturn;
+  onSubmit: (data: FieldValues) => void;
   userAddress: string;
+  defaultValues?: FieldValues;
 }) {
-  const { register, setValue, control, watch } = form;
-  const formData = watch();
-  const isValidFromNeeded = formData.isValidFromNeeded;
-  const { assets } = useSafeBalances();
+  const form = useForm<typeof orderOverviewSchema._type>({
+    resolver: zodResolver(orderOverviewSchema),
+    mode: "onSubmit",
+  });
+  const {
+    register,
+    setValue,
+    clearErrors,
+    formState: { errors },
+  } = form;
 
-  const tokenSell = assets.find(
-    (asset) => asset.tokenInfo.address === formData.tokenSell?.address,
+  useEffect(() => {
+    register("tokenBuy");
+    register("tokenSell");
+    register("validFrom");
+  }, []);
+
+  const { assets, loaded } = useSafeBalances();
+
+  function getTokenBalanceFromAddress(address: string | undefined) {
+    if (!address) return;
+    return assets.find(
+      (tokenBalance) => tokenBalance.tokenInfo.address === address,
+    );
+  }
+
+  const [isValidFromNeeded, setIsValidFromNeeded] = useState(
+    !!defaultValues?.validFrom,
   );
+  const [tokenBuy, setTokenBuy] = useState<TokenBalance | undefined>();
+  const [tokenSell, setTokenSell] = useState<TokenBalance | undefined>();
```

Comment:
would remove this

suggestion here: https://github.com/bleu-fi/monorepo/pull/473